sdk/identity/identity-spa/README.md

```diff
@@ -0,0 +1,228 @@
+      await credential.authenticate(scope);
+    }
+  }
+}
+
+main();
+```
+ 
+### Prerequisites
+
+- An [Azure subscription](https://azure.microsoft.com/free/).
+- An app registration configured with a redirect endpoint of the type `spa`.
+
+To configure the Azure App Registration through the Azure Portal:
+
+- [Register a single page application](https://docs.microsoft.com/azure/active-directory/develop/scenario-spa-app-registration) in the Microsoft identity platform
+- Configure the app registration with a redirect URI to specify where the Microsoft identity platform should redirect the client along with any security tokens.
+  - Follow the instructions at [Redirect URI: MSAL.js 2.0 with auth code flow](https://docs.microsoft.com/azure/active-directory/develop/scenario-spa-app-registration#redirect-uri-msaljs-20-with-auth-code-flow)
+- Ensure that your application has the correct permission for the APIs it intends to use.
+  - In your app registration in the Azure portal, go to `API Permissions`
+  - Click on `Add a permission`
+  - Select the API you want to use. For example, if you're using any of our management/control plane packages (the ones whose name starts with `@azure/arm-`), you should select **Azure Service Management**.
+- Ensure that your AAD Application has enabled public authentication flows:
+  - Go to Azure Active Directory in the Azure portal and find your app registration.
+  - Navigate to the **Authentication** section.
+  - Under **Advanced settings**, select **yes** on the option **Allow public client flows**.
+
+### Install the package
+
+```sh
+$ npm install --save-dev @azure/identity-spa
+```
+
+#### Supported Environments
+
+Azure Identity plugins for JavaScript support stable (even numbered) versions of Node.js starting from v12. While the plugins may run in other Node versions, no support is guaranteed. `@azure/identity-spa` **does not** support browser environments.
+
+## Key concepts
+
+If this is your first time using `@azure/identity` or the Microsoft identity platform (Azure Active Directory), we recommend that you read [Using `@azure/identity` with Microsoft Identity Platform](https://github.com/Azure/azure-sdk-for-js/blob/main/documentation/using-azure-identity.md) first. This document will give you a deeper understanding of the platform and how to configure your Azure account correctly.
+
+## Examples
+
+An example using the `RedirectCredential`:
+
+```javascript
+const { RedirectCredential } = require("@azure/identity-spa");
+const { KeyClient } = require("@azure/keyvault-keys");
+
+async function main() {
+  // Within a web page...
+  const credential = new RedirectCredential({
+    clientId,
+    redirectUri
+  });
+
+  // To log out a previously authenticated user...
+  let authenticationRecord = window.localStorage.authenticationRecord;
+  if (authenticationRecord) {
+    await credential.logout(JSON.parse(authenticationRecord));
+  }
```

Comment:
```suggestion
  let authenticationRecord
  try {
    authenticationRecord = await credential.getAuthenticationRecord();
  } catch(e) {
    // `getAuthenticationRecord` will throw if the user hasn’t authenticated.
    if (e.name === "AuthenticationRequiredError") {
      // Redirection will be triggered by the authenticate() method.
      await credential.authenticate(scope, {
        // The authenticate() method supports sending the browser application state.
        state: JSON.stringify(state)
      });
    }
  }
```

Reply:
Decided not to go this way. We can add `getAuthenticationRecord` later on request.